aten/src/ATen/functorch/BatchRulesLinearAlgebra.cpp

```diff
@@ -280,6 +280,61 @@ static void expect_at_least_rank(
       rank, " dimensions instead.");
 }
 
+twoOutputs _scaled_dot_product_efficient_attention_batch_rule(
+  const Tensor& query, optional<int64_t> query_bdim,
+  const Tensor& key, optional<int64_t> key_bdim,
+  const Tensor& value, optional<int64_t> value_bdim,
+  bool compute_log_sumexp,
+  bool is_causal,
+  c10::optional<double> scale
+) {
+  // const auto query_ = moveBatchDimToFront(query, query_bdim);
+  // const auto key_ = moveBatchDimToFront(key, key_bdim);
+  // const auto value_ = moveBatchDimToFront(value, value_bdim);
+
+  const auto query_ = reshape_dim_into(*query_bdim, 0, query);
+  const auto key_ = reshape_dim_into(*key_bdim, 0, key);
+  const auto value_ = reshape_dim_into(*value_bdim, 0, value);
```

Comment:
it's possible that the bdims don't exist. This can happen in a situation where we are running vmap over some inputs but not others and we would need to handle this case

Reply:
ahh that makes sense! Can I rely on either all the bdims being existent (or non-existent) together?

Reply:
No, there might be a mix. You can use `ensure_has_bdim` to take a Tensor and expand out the bdim though, but this sometimes comes at the cost of performance

Reply:
ok! i will look at other functions to see what the best practice is and try to move in a fix